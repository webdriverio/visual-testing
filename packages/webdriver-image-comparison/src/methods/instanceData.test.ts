import { describe, it, expect, vi } from 'vitest'
import getEnrichedInstanceData from './instanceData.js'

describe('getEnrichedInstanceData', () => {
    it('should be able to enrich the instance data with all the defaults for desktop with no shadow padding', async () => {
        const instanceOptions = {
            addressBarShadowPadding: 6,
            toolBarShadowPadding: 6,
            browserName: 'browserName',
            browserVersion: 'browserVersion',
            deviceName: 'deviceName',
            logName: 'logName',
            name: 'name',
            nativeWebScreenshot: false,
            platformName: 'platformName',
            platformVersion: 'platformVersion',
        }
        const MOCKED_EXECUTOR = vi
            .fn()
            // getEnrichedInstanceData for: getScreenDimensions
            .mockResolvedValueOnce({
                body: {
                    offsetHeight: 0,
                    scrollHeight: 0,
                },
                html: {
                    clientHeight: 0,
                    clientWidth: 0,
                    offsetHeight: 0,
                    scrollHeight: 0,
                    scrollWidth: 0,
                },
                window: {
                    devicePixelRatio: 1,
                    isEmulated: false,
                    innerHeight: 768,
                    innerWidth: 1024,
                    outerHeight: 768,
                    outerWidth: 1024,
                    screenHeight: 0,
                    screenWidth: 0,
                },
            })

        expect(await getEnrichedInstanceData(MOCKED_EXECUTOR, instanceOptions, false)).toMatchSnapshot()
    })

    it('should be able to enrich the instance data with all the defaults for Android ChromeDriver with no shadow padding', async () => {
        const instanceOptions = {
            addressBarShadowPadding: 6,
            toolBarShadowPadding: 6,
            browserName: 'browserName',
            browserVersion: 'browserVersion',
            deviceName: 'deviceName',
            logName: 'logName',
            name: 'name',
            nativeWebScreenshot: false,
            platformName: 'Android',
            platformVersion: '8.0',
        }
        const MOCKED_EXECUTOR = vi
            .fn()
            // getEnrichedInstanceData for: getScreenDimensions
            .mockResolvedValueOnce({
                body: {
                    offsetHeight: 0,
                    scrollHeight: 0,
                },
                html: {
                    clientHeight: 0,
                    clientWidth: 0,
                    offsetHeight: 0,
                    scrollHeight: 0,
                    scrollWidth: 0,
                },
                window: {
                    devicePixelRatio: 1,
                    isEmulated: false,
                    innerHeight: 768,
                    innerWidth: 1024,
                    outerHeight: 768,
                    outerWidth: 1024,
                    screenHeight: 0,
                    screenWidth: 0,
                },
            })

        expect(await getEnrichedInstanceData(MOCKED_EXECUTOR, instanceOptions, false)).toMatchSnapshot()
    })

    it('should be able to enrich the instance data with all the defaults for Android Native Webscreenshot with no shadow padding', async () => {
        const instanceOptions = {
            addressBarShadowPadding: 6,
            toolBarShadowPadding: 6,
            browserName: 'browserName',
            browserVersion: 'browserVersion',
            deviceName: 'deviceName',
            logName: 'logName',
            name: 'name',
            nativeWebScreenshot: true,
            platformName: 'Android',
            platformVersion: '8.0',
        }
        const MOCKED_EXECUTOR = vi
            .fn()
            // getEnrichedInstanceData for: getScreenDimensions
            .mockResolvedValueOnce({
                body: {
                    offsetHeight: 0,
                    scrollHeight: 0,
                },
                html: {
                    clientHeight: 0,
                    clientWidth: 0,
                    offsetHeight: 0,
                    scrollHeight: 0,
                    scrollWidth: 0,
                },
                window: {
                    devicePixelRatio: 1,
                    isEmulated: false,
                    innerHeight: 768,
                    innerWidth: 1024,
                    outerHeight: 768,
                    outerWidth: 1024,
                    screenHeight: 0,
                    screenWidth: 0,
                },
            })

        expect(await getEnrichedInstanceData(MOCKED_EXECUTOR, instanceOptions, false)).toMatchSnapshot()
    })

    it('should be able to enrich the instance data with all the defaults for iOS with shadow padding', async () => {
        const instanceOptions = {
            addressBarShadowPadding: 6,
            toolBarShadowPadding: 6,
            browserName: 'browserName',
            browserVersion: 'browserVersion',
            deviceName: 'deviceName',
            logName: 'logName',
            name: 'name',
            nativeWebScreenshot: false,
            platformName: 'iOS',
            platformVersion: '12.4',
        }
        const MOCKED_EXECUTOR = vi
            .fn()
            // getEnrichedInstanceData for: getScreenDimensions
            .mockResolvedValueOnce({
                body: {
                    offsetHeight: 0,
                    scrollHeight: 0,
                },
                html: {
                    clientHeight: 0,
                    clientWidth: 0,
                    offsetHeight: 0,
                    scrollHeight: 0,
                    scrollWidth: 0,
                },
                window: {
                    devicePixelRatio: 1,
                    isEmulated: false,
                    innerHeight: 768,
                    innerWidth: 1024,
                    outerHeight: 768,
                    outerWidth: 1024,
                    screenHeight: 0,
                    screenWidth: 0,
                },
            })

        expect(await getEnrichedInstanceData(MOCKED_EXECUTOR, instanceOptions, true)).toMatchSnapshot()
    })
})
