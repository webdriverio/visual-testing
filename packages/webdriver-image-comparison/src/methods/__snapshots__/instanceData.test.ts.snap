// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getEnrichedInstanceData > should be able to enrich the instance data with all the defaults for Android ChromeDriver with no shadow padding 1`] = `
{
  "addressBarShadowPadding": 0,
  "body": {
    "offsetHeight": 0,
    "scrollHeight": 0,
  },
  "browserName": "browserName",
  "browserVersion": "browserVersion",
  "deviceName": "deviceName",
  "html": {
    "clientHeight": 0,
    "clientWidth": 0,
    "offsetHeight": 0,
    "scrollHeight": 0,
    "scrollWidth": 0,
  },
  "isAndroid": true,
  "isAndroidChromeDriverScreenshot": true,
  "isAndroidNativeWebScreenshot": false,
  "isIOS": false,
  "isMobile": true,
  "isTestInBrowser": true,
  "isTestInMobileBrowser": true,
  "logName": "logName",
  "name": "name",
  "nativeWebScreenshot": false,
  "platformName": "Android",
  "platformVersion": "8.0",
  "toolBarShadowPadding": 0,
  "window": {
    "devicePixelRatio": 1,
    "innerHeight": 768,
    "innerWidth": 1024,
    "isEmulated": false,
    "outerHeight": 768,
    "outerWidth": 1024,
    "screenHeight": 0,
    "screenWidth": 0,
  },
}
`;

exports[`getEnrichedInstanceData > should be able to enrich the instance data with all the defaults for Android Native Webscreenshot with no shadow padding 1`] = `
{
  "addressBarShadowPadding": 0,
  "body": {
    "offsetHeight": 0,
    "scrollHeight": 0,
  },
  "browserName": "browserName",
  "browserVersion": "browserVersion",
  "deviceName": "deviceName",
  "html": {
    "clientHeight": 0,
    "clientWidth": 0,
    "offsetHeight": 0,
    "scrollHeight": 0,
    "scrollWidth": 0,
  },
  "isAndroid": true,
  "isAndroidChromeDriverScreenshot": false,
  "isAndroidNativeWebScreenshot": true,
  "isIOS": false,
  "isMobile": true,
  "isTestInBrowser": true,
  "isTestInMobileBrowser": true,
  "logName": "logName",
  "name": "name",
  "nativeWebScreenshot": true,
  "platformName": "Android",
  "platformVersion": "8.0",
  "toolBarShadowPadding": 0,
  "window": {
    "devicePixelRatio": 1,
    "innerHeight": 768,
    "innerWidth": 1024,
    "isEmulated": false,
    "outerHeight": 768,
    "outerWidth": 1024,
    "screenHeight": 0,
    "screenWidth": 0,
  },
}
`;

exports[`getEnrichedInstanceData > should be able to enrich the instance data with all the defaults for desktop with no shadow padding 1`] = `
{
  "addressBarShadowPadding": 0,
  "body": {
    "offsetHeight": 0,
    "scrollHeight": 0,
  },
  "browserName": "browserName",
  "browserVersion": "browserVersion",
  "deviceName": "deviceName",
  "html": {
    "clientHeight": 0,
    "clientWidth": 0,
    "offsetHeight": 0,
    "scrollHeight": 0,
    "scrollWidth": 0,
  },
  "isAndroid": false,
  "isAndroidChromeDriverScreenshot": false,
  "isAndroidNativeWebScreenshot": false,
  "isIOS": false,
  "isMobile": false,
  "isTestInBrowser": true,
  "isTestInMobileBrowser": false,
  "logName": "logName",
  "name": "name",
  "nativeWebScreenshot": false,
  "platformName": "platformName",
  "platformVersion": "platformVersion",
  "toolBarShadowPadding": 0,
  "window": {
    "devicePixelRatio": 1,
    "innerHeight": 768,
    "innerWidth": 1024,
    "isEmulated": false,
    "outerHeight": 768,
    "outerWidth": 1024,
    "screenHeight": 0,
    "screenWidth": 0,
  },
}
`;

exports[`getEnrichedInstanceData > should be able to enrich the instance data with all the defaults for iOS with shadow padding 1`] = `
{
  "addressBarShadowPadding": 6,
  "body": {
    "offsetHeight": 0,
    "scrollHeight": 0,
  },
  "browserName": "browserName",
  "browserVersion": "browserVersion",
  "deviceName": "deviceName",
  "html": {
    "clientHeight": 0,
    "clientWidth": 0,
    "offsetHeight": 0,
    "scrollHeight": 0,
    "scrollWidth": 0,
  },
  "isAndroid": false,
  "isAndroidChromeDriverScreenshot": false,
  "isAndroidNativeWebScreenshot": false,
  "isIOS": true,
  "isMobile": true,
  "isTestInBrowser": true,
  "isTestInMobileBrowser": true,
  "logName": "logName",
  "name": "name",
  "nativeWebScreenshot": false,
  "platformName": "iOS",
  "platformVersion": "12.4",
  "toolBarShadowPadding": 15,
  "window": {
    "devicePixelRatio": 1,
    "innerHeight": 768,
    "innerWidth": 1024,
    "isEmulated": false,
    "outerHeight": 768,
    "outerWidth": 1024,
    "screenHeight": 0,
    "screenWidth": 0,
  },
}
`;
